tmpl/wiki/tutorial-lwt.md

```diff
@@ -74,508 +111,327 @@ Assuming you are building for POSIX using native kernel sockets, compile the app
+If you want to spawn a thread without waiting for the result, use `Lwt.async`:
 
 ```
-  open V1_LWT
-  open Lwt
-  open OS
-
-  module Main (C : CONSOLE) = struct
-    let start c =
-
-      let heads =
-        Time.sleep 1.0 >>
-        return (C.log c "Heads");
-      in
-      let tails =
-        Time.sleep 2.0 >>
-        return (C.log c "Tails");
-      in
-      lwt () = heads <&> tails in
-      C.log c "Finished";
-      return ()
-  end
+Lwt.async (fun () ->
+  OS.Time.sleep 10.0 >>= fun () ->
+  C.log_s c "Finished"
+)
 ```
 
-This is in `lwt/src/unikernels.ml` in the MirageOS code repository. The target is `heads3`.
-
-Here we define two threads, `heads` and `tails`, and block until they are both complete (via the `lwt ()` and the `<&>` join operator). If you want to print "Finished" before the previous threads are complete, just put the print statement (`Console.log`) before the join statement (`... <&> ...`).
-
+**Note**: do _not_ do `let _ = my_background_thread ()`. This ignores the result of the thread, which means that if it fails with an exception then the error will never be reported.
 
-##Cancelling
+`Lwt.async` reports errors to the user's configured `Lwt.async_exception_handler`, which may or may not terminate the unikernel depending on how it has been configured.
 
-In order to cancel a thread, the function `cancel` (provided by the module Lwt) is needed. It has type `'a t -> unit` and does exactly what it says (except on certain complicated cases that are not in the scope of this tutorial). A simple timeout function that cancels a thread after a given number of seconds can be written easily:
+It is often better to catch such exceptions and log them with some contextual information.
+Here's some real Mirage code that spawns a new background thread to handle a new frame received from the network.
+The log message includes the exception it caught, a dump of the troublesome frame and, like all log messages, information about when it occured and in which module.
 
 ```
-  (* In this examples and all those afterwards, we consider Lwt and OS to be
-     opened *)
-  let timeout f t =
-    Time.sleep f >>= fun () -> cancel t
+(* Handle a frame of data from the network... *)
+Lwt.async (fun () ->
+  Lwt.catch (fun () -> fn data)
+    (fun ex ->
+       Log.err (fun f -> f "uncaught exception from listen callback \
+                            while handling frame:@\n%a@\nException: @[%s@]"
+                   S.pp_frame data (Printexc.to_string ex));
+       Lwt.return ()
+    )
+)
 ```
 
-###Challenge
+By the way, the reason `async` and `catch` take functions rather than plain threads is so they can catch OCaml exceptions (not just failed threads). Otherwise, if we had `Lwt.catch (fn data) ...` and `fn data` used `raise` then the exception would occur before `Lwt.catch` ran, and it wouldn't be able to handle it.
```

Comment:
This paragraph might be easier to understand if it explained the more general reason: "to have control over when the threads are executed. A thread wrapped in a function won't start until the function is called. This allows `async` and `catch` to set up exception handlers."

That is quite a bit less precise, but I think easier for a beginner to understand. And, it gives the right intuition for other uses of threads wrapped in functions. For example, as I'm sure you know, you might have some function that sets up an environment, then runs a user-provided thread, then cleans up the environment. You wouldn't want the user's thread to potentially run before the environment is ready. Maybe you can mention this here very briefly, ("this trick is also used in other places..."), but I don't think it's necessary.